main.py

```diff
@@ -0,0 +1,83 @@
+import os
+from typing import Any, Dict
+
+from dotenv import load_dotenv
+from fastapi import FastAPI, HTTPException
+from openai import OpenAI
+from pydantic import BaseModel
+
+from pseudoanonymize.anonymization import Anonymizer
+from pseudoanonymize.deanonymization import Deanonymizer
+from pseudoanonymize.exceptions import MaxRetriesExceededException
+
+app = FastAPI()
+
+load_dotenv()
+openai_api_key = os.getenv("OPENAI_API_KEY")
+client = OpenAI(api_key=openai_api_key)
+
+
+anonymizer = Anonymizer(client=client)
+deanonymizer = Deanonymizer(client=client)
+
+
+class AnonymizeRequest(BaseModel):
+    text: str
+    retries: int = 5
+
+
+class DeanonymizeRequest(BaseModel):
+    text: str
+    replacement_dict: Dict[Any, str]
+    retries: int = 5
+
+
+def flatten_replacement_dict(replacement_dict):
+    flat_dict = {}
+    for key, value in replacement_dict.items():
+        if isinstance(key, tuple):
+            for sub_key in key:
+                flat_dict[sub_key] = value
+        else:
+            flat_dict[key] = value
+    return flat_dict
+
+
+@app.post("/anonymize")
+async def anonymize(request: AnonymizeRequest):
+    try:
+        prediction = anonymizer.retry_prediction({"text": request.text}, request.retries)
```

Comment:
Why are we passing the request text within a dictionary? Why not pass it as a parameter `retry_prediction(text=request.text)`?

Reply:
It's not ideal, but I did this to have the predict method for the anon/deanon share the same signature.   

```
    @abstractmethod
    def predict(self, input_: Dict[str, Any]) -> Dict[str, Any]:
        pass
```
I needed this because the anon predict method only takes an input text, but the deanon predict method needs the text and a replacement_dict. 

Perhaps a better approach was to do something like this?
```
    @abstractmethod
    def predict(self, input_: str, **kwargs) -> Dict[str, Any]:
        pass
 ```

Reply:
I see hmm...

When you run into an issue like this is probably suggests that these two things shouldn't share a base class.

The shared functionality that they currently have in the form of methods on the base class could equally just be utils functions that each class uses.

Or... if you like to use classes you could create a separate `BaseAnonymizer` and `BaseDeanonymizer` that are free to have their own predict methods with their own arguments.